### PR TITLE
Skip phub for Full and Online media

### PR DESCRIPTION
**Pull request by SergioAtSUSE**
## Observation
- Full media doesn't use SCC, not internet connection. Not possible to
get packages from that module.
- Online media fails to add package hub repository because it is missing
for SLE15-SP2.

## Related tickets
- https://bugzilla.suse.com/show_bug.cgi?id=1157659
- https://bugzilla.suse.com/show_bug.cgi?id=1151373
- https://progress.opensuse.org/issues/60041

## Verification runs
- Online: http://openqa.slindomansilla-vm.qa.suse.de/tests/1998#step/scc_registration/20
- Full: http://openqa.slindomansilla-vm.qa.suse.de/tests/1999#step/addon_products_sle/51

### CODE REVIEW

File: lib/registration.pm [hunk 1]
```diff
@@ -366,6 +367,18 @@ sub _yast_scc_addons_handler {
...
 
+sub skip_package_hub_if_necessary {
+    my ($addon) = @_;
+    if (is_sle('15-SP2+') && $addon eq 'phub') {
+        if (check_var('FLAVOR', 'Online')) {
+            record_soft_failure('bsc#1151373 - Missing or broken repository after registering module PackageHub');
+        } elsif (check_var('FLAVOR', 'Full')) {
+            record_info('Full media: no PHUB', 'Skipping Package Hub, it is not available on offline scenarios - bsc#1157659');
+        }
+        next;
```
Comment (rwx788) [inline lib/registration.pm]: I'm not sure if that's going to work. We need this `next` statement in the loop, so I would return true when we need to skip it.
```perl
next if skip_package_hub_if_necessary($addon);
```

Comment (SergioAtSUSE) [reply lib/registration.pm]: In the two verification runs it works

Comment (rwx788) [reply lib/registration.pm]: True... Perl is weird, I would not expect `next` to exit loop outside of subroutine. @foursixnine is that ok? =)

Comment (rwx788) [reply lib/registration.pm]: Anyway, let's merge to unblock testing.

**Pull request merged**